### PR TITLE
Feature name channel translation in slack message text

### PR DESCRIPTION
**Pull request by drewpearce**
Slack does funny things with user and channel references. this seeks to automatically encode/decode them in the slack connector (in the message text) to that the message text is usable for addressing these items.

### CODE REVIEW

File: Legobot/Connectors/Slack.py [hunk 1]
```diff
@@ -102,14 +109,140 @@ def run(self):
...
+           display names.
+
+        Args:
+            text (string): The message text
+        Returns:
+            string: The message text with userids replaced.
+        '''
+
+        match = True
+        pattern = re.compile('<@([A-Z0-9]+)>')
```
Comment (bbriggs) [inline Legobot/Connectors/Slack.py]: Should this be length delimited?

File: Legobot/Connectors/Slack.py [hunk 2]
```diff
@@ -102,14 +109,140 @@ def run(self):
...
+           replace them with just channel names.
+
+        Args:
+            text (string): The message text
+        Returns:
+            string: The message text with channel references replaced.
+        '''
+
+        match = True
+        pattern = re.compile('<#([A-Z0-9]+)\|([A-Za-z0-9-]+)>')
```
Comment (bbriggs) [inline Legobot/Connectors/Slack.py]: Should this be length delimited?

File: Legobot/Connectors/Slack.py [hunk 3]
```diff
@@ -40,8 +41,7 @@ class RtmBot(threading.Thread, object):
         slack_client: initialized RTM client
     """
 
-    def __init__(self, baseplate, token, actor_urn, reconnect=True,
-                 *args, **kwargs):
+    def __init__(self, baseplate, token, actor_urn, *args, **kwargs):
```
[the remark this answers is not in the document]
Comment (drewpearce) [reply Legobot/Connectors/Slack.py]: finally figured out what was happening. this is an outdated change. it no longer applies

File: Legobot/Connectors/Slack.py [hunk 4]
```diff
@@ -59,16 +59,13 @@ def __init__(self, baseplate, token, actor_urn, reconnect=True,
         self.token = token
         self.last_ping = 0
         self.actor_urn = actor_urn
-        self.reconnect = reconnect
```
[the remark this answers is not in the document]
Comment (drewpearce) [reply Legobot/Connectors/Slack.py]: finally figured out what was happening. this is an outdated change. it no longer applies

File: Legobot/Connectors/Slack.py (hunk 1 above)
Comment (drewpearce) [reply Legobot/Connectors/Slack.py]: resolved in latest commit

File: Legobot/Connectors/Slack.py (hunk 2 above)
Comment (drewpearce) [reply Legobot/Connectors/Slack.py]: resolved in latest commit

**Pull request merged**